lang/test/org/partiql/lang/eval/EvaluatingCompilerTests.kt

```diff
@@ -1475,4 +1535,68 @@ class EvaluatingCompilerTests : EvaluatorTestBase() {
                      listOf("a", "b"),
                      result.first().orderedNames)
     }
+
+    @Test
+    fun selectDistinct() = assertEval(
+        """SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t""",
+        """
+          [{a: 1}, {a: 2}]
+        """)
+
+    @Test
+    fun selectDistinctWithAggregate() = assertEval(
+        """SELECT SUM(DISTINCT t.a) AS a FROM `[{a:10}, {a:1}, {a:10}, {a:3}]` t""",
+        "[{a:14}]")
+
+    @Test
+    fun selectDistinctSubQuery() = assertEval(
+        """SELECT * FROM (SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t)""", 
+        """[{a:1},{a:2}]""")
+    
+    @Test
+    fun selectDistinctWithGroupBy() = assertEval(
+        """
+            SELECT t.a, COUNT(DISTINCT t.b) AS c 
+            FROM `[{a:1, b:10}, {a:1, b:10}, {a:1, b:20}, {a:2, b:10}, {a:2, b:10}]` t
+            GROUP by t.a
```

Comment:
Speaking of `GROUP BY` we need a bunch of tests which include the use of `SELECT DISTINCT` and `GROUP BY` together.  Postgres supports this:  http://sqlfiddle.com/#!17/0daa8/4

Reply:
I didn't add that test case as the `distinct` keyword should be a no-op, but you are right it's good to have to show it doesn't change behavior in any way